src/courses/advanced/05.md

```diff
@@ -412,10 +412,10 @@ class Git < Inspec.resource(1)
     end
 
     def current_branch
```

Comment:
I think both of the implementations for these branch methods are wrong btw.

branches should be something like the following but using inspec.command(thing).stdout instead of directly calling out to the shell and also using the --git-dir flag: 
```ruby
`git branch --format='%(refname:short)'`.lines.map(&:strip)
```

same same but different with current branch:
```ruby
`git branch --show-current`.strip
```

`includes?` is a function on string which is why branches works as it does with the control being `should include 'main'` but in actuality this means that we could potentially be passing even when the list of branches didn't include a branch exactly called `main` - ex. if it were like something `mainline` then it would still pass.  doing it with the alternative implementation above where we convert it into an array makes us use the array version of `includes?` which looks for exact string comparison amongst its members

the current branch implementation is what yall been saying where you examine all possibilities vs filter precisely to what we're looking for.  turns out `git branch` has a flag that makes what we wanna do pretty damn easy lol

Reply:
yes :)